tests/test_flow/test_api.py

```diff
@@ -376,6 +381,36 @@ def test_get_formats(preset_flow):
         assert list(p_series_index_df['q']) == [5]
 
 
+def test_get_modes_exclude_object(preset_flow):
+    # TODO: should we raise a warning for ValueProvider entities when
+    # queried for a mode that implies the existence of a persisted file?
+    flow = preset_flow
+
+    for mode in [Path, 'path']:
+        y_path = flow.get('y', mode=mode)
+        assert y_path is None
+
+        y_fxn_path = flow.get('y_fxn', mode=mode)
+        assert isinstance(y_fxn_path, Path)
+        assert y_fxn_path.is_file()
+
+    y_filecopier = flow.get('y', mode='FileCopier')
+    assert isinstance(y_filecopier, bn.util.FileCopier)
+    assert y_filecopier.src_file_path is None
+
+    y_fxn_filecopier = flow.get('y_fxn', mode='FileCopier')
+    assert isinstance(y_fxn_filecopier, bn.util.FileCopier)
+    assert y_fxn_filecopier.src_file_path.is_file()
```

Comment:
Is the idea that we don't need to test the actual value because we're doing the "real" test in `test_copy`?  Maybe we could do the full test here (i.e., load the value and check that it's correct) but keep the more complicated combinations in `test_copy`?

Reply:
Sure, added: `pickle.loads(y_fxn_filecopier.src_file_path.read_bytes()) == 1`

Reply:
If it were me I would just copy the file (so the test matches the expected user behavior) but I think this is fine too.